Tests/MongoSwiftTests/ConnectionStringTests.swift

```diff
@@ -203,13 +155,49 @@ final class ConnectionStringTests: MongoSwiftTestCase {
                 }
 
                 // Assert that auth matches, if present
-//                if let expectedAuth = testCase.auth {
-//                    let actual = connString.credential
-//                    guard expectedAuth.matches(actual) else {
-//                        XCTFail("Expected credentials: \(expectedAuth) do not match parsed credentials: \(actual)")
-//                        continue
-//                    }
-//                }
+                if let expectedAuth = testCase.auth {
+                    let actual = connString.credential ?? MongoCredential()
+                    guard expectedAuth.matches(actual) else {
+                        print(testCase.description)
+                        XCTFail("Expected credentials: \(expectedAuth) do not match parsed credentials: \(actual)")
+                        continue
+                    }
+                }
+
+                if let expectedOptions = testCase.options {
```

Comment:
I'll probably revisit this implementation once I have a better handle on all of the options that need to be checked/tested.